core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -147,7 +150,9 @@ abstract class SparkFunSuite
     if (excluded.contains(testName)) {
       ignore(s"$testName (excluded)")(testBody)
     } else {
-      super.test(testName, testTags: _*)(testBody)
+      super.test(testName, testTags: _*)(
+        failAfter(10.minutes)(testBody)
```

Comment:
@zhengruifeng Like this, I gave a simple test

For example, change `failAfter(10.minutes)(testBody)` to `failAfter(100.millis)(testBody)`, then run a test like
```
  test("timeout") {
    Thread.sleep(2000)
  }
```

then 

```
[info] - timeout *** FAILED *** (2 seconds, 35 milliseconds)
[info]   The code passed to failAfter did not complete within 100 milliseconds. (DistributionSuite.scala:27)
[info]   org.scalatest.exceptions.TestFailedDueToTimeoutException:
[info]   at java.lang.Thread.getStackTrace(Thread.java:1564)
```

and other test will continue to execute

Reply:
I am not sure whether to add the `timeout` in general, or only used in the slowest ones

also cc @dongjoon-hyun @HyukjinKwon